raft/node.go

```diff
@@ -261,6 +265,9 @@ type node struct {
 	stop       chan struct{}
 	status     chan chan Status
 
+	id       func() uint64
```

Comment:
Let's just put the`*RawNode` here. Then we can also remove `logger` and you don't need to pass the RawNode to `run` any more. The current setup is quite confusing.